protocols/mdns/src/behaviour/iface.rs

```diff
@@ -159,6 +186,25 @@ where
             if Pin::new(&mut self.timeout).poll_next(cx).is_ready() {
                 log::trace!("sending query on iface {}", self.addr);
                 self.send_buffer.push_back(build_query());
+                log::trace!("tick on {:#?} {:#?}", self.addr, self.probe_state);
+
+                let continue_probing = self.discovered.is_empty();
```

Comment:
Do we need this here if we set it on line 268 already?

I think it is easier to understand if we pair the changes to `probe_state` with the changes to `discovered`.